dns_sd_avahi.c

```diff
@@ -255,3 +283,88 @@ int dnssd_find_hosts(struct dns_sd_discovery_data **ddata)
 
 	return ret;
 }
+
+void __avahi_resolve_host(struct dns_sd_discovery_data *d, const char *hostname,
+			  const AvahiProtocol proto)
+{
+	AvahiClient *client;
+	AvahiHostNameResolver *resolver;
+	int ret;
+
+	d->poll = avahi_simple_poll_new();
+	if (!d->poll)
+		return;
+
+	client = avahi_client_new(avahi_simple_poll_get(d->poll), 0, NULL, NULL, &ret);
+	if (!client) {
+		IIO_ERROR("Unable to create Avahi DNS-SD client :%s\n", avahi_strerror(ret));
+		goto err_free_poll;
+	}
+
+	resolver = avahi_host_name_resolver_new(client, AVAHI_IF_UNSPEC, proto, hostname, proto, 0,
+						__avahi_host_resolver, d);
+	if (!resolver) {
+		ret = avahi_client_errno(client);
+		IIO_ERROR("Unable to create Avahi DNS-SD browser: %s\n", avahi_strerror(ret));
+		goto err_free_client;
+	}
+
+	IIO_DEBUG("Trying to resolve host: %s\n, proto: %d", hostname, proto);
+	avahi_simple_poll_loop(d->poll);
+
+err_free_client:
+	avahi_client_free(client);
+err_free_poll:
+	avahi_simple_poll_free(d->poll);
+}
+
+int dnssd_resolve_host(const char *hostname, char *ip_addr, const int addr_len)
+{
+	struct dns_sd_discovery_data *d;
+	int ret = 0;
+
+	if (!hostname || hostname[0] == '\0')
+		return -EINVAL;
+
+	if (new_discovery_data(&d) < 0)
```

Comment:
i didn't follow the rest of the flow yet; but does `new_discovery_data()` require to be under lock?
or is this in a safe context to let it be unlocked?

Reply:
ah wait, stuff gets allocated there;
nevermind

Reply:
Don't think so...